src/wallets/hardware/LedgerWallet.ts

```diff
@@ -73,6 +66,32 @@ export default class LedgerWallet
     this.derivationPath = path;
   }
 
+  async setAccountId(accountId: string) {
+    this.accountId = accountId;
+  }
+
+  async checkAccountId(accountId: string, publicKey: string) {
+    const state = getState();
+    const provider = new providers.JsonRpcProvider(`https://rpc.${state.options.networkId}.near.org`);
+
+    const publicKeyString = publicKey;
+
+    const response: any = await provider
+      .query({
+        request_type: "view_access_key",
+        finality: "final",
+        account_id: accountId,
+        public_key: publicKeyString,
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+
+    this.nonce = response.nonce;
```

Comment:
This will crash if the `.catch` path is triggered as `response` will be `undefined`